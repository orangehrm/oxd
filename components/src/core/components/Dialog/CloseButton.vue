<template>
  <button class="oxd-dialog-close-button">Ã—</button>
</template>

<script lang="ts">
import {defineComponent} from 'vue';

export default defineComponent({
  name: 'oxd-dialog-close-button',
});
</script>

<style src="./close-button.scss" lang="scss" scoped></style>
