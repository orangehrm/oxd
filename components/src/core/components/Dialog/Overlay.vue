<template>
  <div role="dialog" :class="classes">
    <slot></slot>
  </div>
</template>

<script lang="ts">
import {defineComponent} from 'vue';

export default defineComponent({
  name: 'oxd-overlay',

  props: {
    show: {
      type: Boolean,
      default: false,
    },
    centered: {
      type: Boolean,
      default: true,
    },
  },

  computed: {
    classes(): object {
      return {
        'oxd-overlay': true,
        'oxd-overlay--flex': true,
        'oxd-overlay--flex-centered': this.centered,
        'oxd-overlay--hide': !this.show,
      };
    },
  },
});
</script>

<style src="./overlay.scss" lang="scss" scoped></style>
