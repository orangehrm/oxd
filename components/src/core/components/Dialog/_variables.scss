$oxd-overlay-background-color: rgba($oxd-black-color, 0.5) !default;

$oxd-dialog-close-button-font-color: $oxd-white-color !default;
$oxd-dialog-close-button-bg-color: $oxd-interface-gray-lighten-1-color !default;
$oxd-dialog-close-button-bg-color-hover: $oxd-interface-gray-darken-1-color !default;
$oxd-dialog-close-button-border-radius: 2.5rem !default;
$oxd-dialog-close-button-font-size: 1rem !default;
$oxd-dialog-close-button-font-weight: 600 !default;
$oxd-dialog-close-button-size: 1.5625rem !default;
$oxd-dialog-close-button-top: -0.5rem !default;
$oxd-dialog-close-button-right: -0.5rem !default;

$oxd-dialog-sheet-box-shadow: 0 25px 35px 0 rgba($oxd-black-color, 0.18) !default;
