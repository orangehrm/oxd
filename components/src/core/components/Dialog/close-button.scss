// This file is part of OrangeHRM Inc
//
// Copyright (C) 2020 onwards OrangeHRM Inc
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see  http://www.gnu.org/licenses

@import '../../../styles';
@import 'variables';

.oxd-dialog-close-button {
  cursor: pointer;
  padding: 0;
  border: 0;
  outline: 0;
  display: flex;
  flex-direction: column;
  justify-content: center;
  align-items: center;
  letter-spacing: normal;
  width: $oxd-dialog-close-button-size;
  height: $oxd-dialog-close-button-size;
  color: $oxd-dialog-close-button-font-color;
  font-size: $oxd-dialog-close-button-font-size;
  font-weight: $oxd-dialog-close-button-font-weight;
  border-radius: $oxd-dialog-close-button-border-radius;
  background-color: $oxd-interface-gray-lighten-1-color;

  &:hover {
    background-color: $oxd-interface-gray-darken-1-color;
  }
}
