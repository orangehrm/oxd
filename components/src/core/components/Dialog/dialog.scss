// This file is part of OrangeHRM Inc
//
// Copyright (C) 2020 onwards OrangeHRM Inc
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see  http://www.gnu.org/licenses

@import '../../../styles';
@import 'variables';

.oxd-dialog-close-button-position {
  position: absolute;
  top: $oxd-dialog-close-button-top;
  right: $oxd-dialog-close-button-right;
}

.oxd-dialog-sheet {
  z-index: $oxd-dialog-z-index;
  position: relative;
  box-sizing: border-box;

  &--shadow {
    box-shadow: $oxd-dialog-sheet-box-shadow;
  }
}

.fade-enter-active,
.fade-leave-active {
  transition: opacity 0.25s ease;
}

.fade-enter-from,
.fade-leave-to {
  opacity: 0;
}
