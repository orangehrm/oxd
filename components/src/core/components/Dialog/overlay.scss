@import '../../../styles';
@import 'variables';

.oxd-overlay {
  position: fixed;
  top: 0;
  left: 0;
  z-index: $oxd-overlay-z-index;
  width: 100%;
  height: 100%;
  outline: 0;
  box-sizing: border-box;
  background-color: $oxd-overlay-background-color;

  &--hide {
    display: none !important;
  }

  &--flex {
    display: flex;
    flex-direction: column;

    &-centered {
      justify-content: center;
      align-items: center;
    }
  }
}
