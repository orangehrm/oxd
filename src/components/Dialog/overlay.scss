// This file is part of OrangeHRM Inc
//
// Copyright (C) 2020 onwards OrangeHRM Inc
//
// This program is free software; you can redistribute it and/or modify
// it under the terms of the GNU General Public License as published by
// the Free Software Foundation; either version 3 of the License, or
// (at your option) any later version.
//
// This program is distributed in the hope that it will be useful,
// but WITHOUT ANY WARRANTY; without even the implied warranty of
// MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
// GNU General Public License for more details.
//
// You should have received a copy of the GNU General Public License
// along with this program.  If not, see  http://www.gnu.org/licenses

@import 'variables';

.oxd-overlay {
  position: fixed;
  top: 0;
  left: 0;
  z-index: $oxd-overlay-z-index;
  width: 100%;
  height: 100%;
  outline: 0;
  box-sizing: border-box;
  background-color: $oxd-overlay-background-color;

  &--hide {
    display: none !important;
  }

  &--flex {
    display: flex;
    flex-direction: column;

    &-centered {
      justify-content: center;
      align-items: center;
    }
  }
}
