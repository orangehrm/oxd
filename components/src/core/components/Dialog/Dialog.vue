<template>
  <transition :name="computedTransition" appear>
    <oxd-overlay role="dialog" centered :show="true" @click="onClickOverlay">
      <!--
    :aria-labelledby="'dialogTitle_' + id"
    :aria-describedby="'dialogDesc_' + id"
    -->
      <component :is="dialogContainer">
        <oxd-sheet
          :class="classes"
          v-bind="$attrs"
          role="document"
          @click="onClickSheet"
        >
          <oxd-dialog-close-button
            v-if="withClose"
            class="oxd-dialog-close-button-position"
            @click="onClose"
          />
          <slot></slot>
        </oxd-sheet>
      </component>
    </oxd-overlay>
  </transition>
</template>

<script lang="ts">
import {defineComponent, onUnmounted} from 'vue';
import Overlay from '@orangehrm/oxd/core/components/Dialog/Overlay.vue';
import CloseButton from '@orangehrm/oxd/core/components/Dialog/CloseButton.vue';
import Sheet from '@orangehrm/oxd/core/components/Sheet/Sheet.vue';

// Containers
import DefaultContainer from '@orangehrm/oxd/core/components/Dialog/Container/Default.vue';

export default defineComponent({
  name: 'oxd-dialog',
  inheritAttrs: false,

  components: {
    'oxd-overlay': Overlay,
    'oxd-dialog-close-button': CloseButton,
    'oxd-sheet': Sheet,

    // Containers
    'oxd-dialog-container-default': DefaultContainer,
  },

  emits: ['update:show'],

  props: {
    shadow: {
      type: Boolean,
      default: true,
    },
    withClose: {
      type: Boolean,
      default: true,
    },
    persistent: {
      type: Boolean,
      default: false,
    },
    dialogContainer: {
      type: String,
      default: 'oxd-dialog-container-default',
    },
    withTransition: {
      type: Boolean,
      default: true,
    },
    dialogSizeClass: {
      type: String,
      default: 'modal-md',
    },
  },

  computed: {
    classes(): object {
      return {
        'oxd-dialog-sheet': true,
        'oxd-dialog-sheet--shadow': this.shadow,
        [`${this.dialogSizeClass}`]: true,
      };
    },
    computedTransition(): string {
      return this.withTransition ? 'fade' : '';
    },
  },

  methods: {
    onClose() {
      this.$emit('update:show', false);
    },
    onClickOverlay() {
      if (!this.persistent) {
        this.$emit('update:show', false);
      }
    },
    onClickSheet(e: Event) {
      e.stopPropagation();
    },
  },

  mounted() {
    const body = document.getElementsByTagName('body')
    if (body) {
      body[0].classList.add('overflow-hidden')
    }
  },

  unmounted() {
    const body = document.getElementsByTagName('body')
    if (body) {
      body[0].classList.remove('overflow-hidden')
    }
  }
});
</script>

<style src="./dialog.scss" lang="scss" scoped></style>
