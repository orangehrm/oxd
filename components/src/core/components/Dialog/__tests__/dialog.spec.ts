import {mount} from '@vue/test-utils';
import Dialog from '@orangehrm/oxd/core/components/Dialog/Dialog.vue';

describe('Dialog > Dialog.vue', () => {
  it('should renders OXD Dialog > Dialog', () => {
    const wrapper = mount(Dialog, {
      props: {},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Dialog > Dialog show', () => {
    const wrapper = mount(Dialog, {
      props: {show: true},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Dialog > Dialog no shadow', () => {
    const wrapper = mount(Dialog, {
      props: {show: true, shadow: false},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Dialog > Dialog witout close', () => {
    const wrapper = mount(Dialog, {
      props: {show: true, withClose: false},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Dialog > Dialog persistent', () => {
    const wrapper = mount(Dialog, {
      props: {show: true, persistent: true},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });

  it('should renders OXD Dialog > Dialog dialogContainer-div', () => {
    const wrapper = mount(Dialog, {
      props: {show: true, dialogContainer: 'div'},
    });
    expect(wrapper.html()).toMatchSnapshot();
  });
});
