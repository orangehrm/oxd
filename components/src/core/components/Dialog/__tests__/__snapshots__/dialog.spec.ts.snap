// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Dialog > Dialog.vue should renders OXD Dialog > Dialog 1`] = `
<transition-stub>
  <div role="dialog" class="oxd-overlay oxd-overlay--flex oxd-overlay--flex-centered">
    <!--
    :aria-labelledby="'dialogTitle_' + id"
    :aria-describedby="'dialogDesc_' + id"
    -->
    <div class="oxd-dialog-container-default">
      <div class="oxd-dialog-container-default--inner">
        <div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--white oxd-dialog-sheet oxd-dialog-sheet--shadow modal-md" role="document"><button type="button" class="oxd-icon-button oxd-dialog-close-button oxd-dialog-close-button-position" flow="top"><i class="oxd-icon oxd-icon--medium bi-x" style="color: rgb(255, 255, 255);"></i></button></div>
      </div>
    </div>
  </div>
</transition-stub>
`;

exports[`Dialog > Dialog.vue should renders OXD Dialog > Dialog dialogContainer-div 1`] = `
<transition-stub>
  <div role="dialog" class="oxd-overlay oxd-overlay--flex oxd-overlay--flex-centered">
    <!--
    :aria-labelledby="'dialogTitle_' + id"
    :aria-describedby="'dialogDesc_' + id"
    -->
    <div>
      <div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--white oxd-dialog-sheet oxd-dialog-sheet--shadow modal-md" show="true" role="document"><button type="button" class="oxd-icon-button oxd-dialog-close-button oxd-dialog-close-button-position" flow="top"><i class="oxd-icon oxd-icon--medium bi-x" style="color: rgb(255, 255, 255);"></i></button></div>
    </div>
  </div>
</transition-stub>
`;

exports[`Dialog > Dialog.vue should renders OXD Dialog > Dialog no shadow 1`] = `
<transition-stub>
  <div role="dialog" class="oxd-overlay oxd-overlay--flex oxd-overlay--flex-centered">
    <!--
    :aria-labelledby="'dialogTitle_' + id"
    :aria-describedby="'dialogDesc_' + id"
    -->
    <div class="oxd-dialog-container-default">
      <div class="oxd-dialog-container-default--inner">
        <div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--white oxd-dialog-sheet modal-md" show="true" role="document"><button type="button" class="oxd-icon-button oxd-dialog-close-button oxd-dialog-close-button-position" flow="top"><i class="oxd-icon oxd-icon--medium bi-x" style="color: rgb(255, 255, 255);"></i></button></div>
      </div>
    </div>
  </div>
</transition-stub>
`;

exports[`Dialog > Dialog.vue should renders OXD Dialog > Dialog persistent 1`] = `
<transition-stub>
  <div role="dialog" class="oxd-overlay oxd-overlay--flex oxd-overlay--flex-centered">
    <!--
    :aria-labelledby="'dialogTitle_' + id"
    :aria-describedby="'dialogDesc_' + id"
    -->
    <div class="oxd-dialog-container-default">
      <div class="oxd-dialog-container-default--inner">
        <div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--white oxd-dialog-sheet oxd-dialog-sheet--shadow modal-md" show="true" role="document"><button type="button" class="oxd-icon-button oxd-dialog-close-button oxd-dialog-close-button-position" flow="top"><i class="oxd-icon oxd-icon--medium bi-x" style="color: rgb(255, 255, 255);"></i></button></div>
      </div>
    </div>
  </div>
</transition-stub>
`;

exports[`Dialog > Dialog.vue should renders OXD Dialog > Dialog show 1`] = `
<transition-stub>
  <div role="dialog" class="oxd-overlay oxd-overlay--flex oxd-overlay--flex-centered">
    <!--
    :aria-labelledby="'dialogTitle_' + id"
    :aria-describedby="'dialogDesc_' + id"
    -->
    <div class="oxd-dialog-container-default">
      <div class="oxd-dialog-container-default--inner">
        <div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--white oxd-dialog-sheet oxd-dialog-sheet--shadow modal-md" show="true" role="document"><button type="button" class="oxd-icon-button oxd-dialog-close-button oxd-dialog-close-button-position" flow="top"><i class="oxd-icon oxd-icon--medium bi-x" style="color: rgb(255, 255, 255);"></i></button></div>
      </div>
    </div>
  </div>
</transition-stub>
`;

exports[`Dialog > Dialog.vue should renders OXD Dialog > Dialog witout close 1`] = `
<transition-stub>
  <div role="dialog" class="oxd-overlay oxd-overlay--flex oxd-overlay--flex-centered">
    <!--
    :aria-labelledby="'dialogTitle_' + id"
    :aria-describedby="'dialogDesc_' + id"
    -->
    <div class="oxd-dialog-container-default">
      <div class="oxd-dialog-container-default--inner">
        <div class="oxd-sheet oxd-sheet--rounded oxd-sheet--gutters oxd-sheet--white oxd-dialog-sheet oxd-dialog-sheet--shadow modal-md" show="true" role="document"></div>
      </div>
    </div>
  </div>
</transition-stub>
`;
